hydra-cluster/test/Test/EndToEndSpec.hs

```diff
@@ -313,11 +320,50 @@ spec = around showLogsOnFailure $ do
                     metrics <- getMetrics n1
                     metrics `shouldSatisfy` ("hydra_head_events" `BS.isInfixOf`)
 
-    describe "hydra-node executable" $
+    describe "hydra-node executable" $ do
       it "display proper semantic version given it is passed --version argument" $ \_ ->
         failAfter 5 $ do
           version <- readCreateProcess (proc "hydra-node" ["--version"]) ""
           version `shouldSatisfy` (=~ ("[0-9]+\\.[0-9]+\\.[0-9]+(-[a-zA-Z0-9]+)?" :: String))
+      fit "hydra-node logs it's command line arguments" $ \_ -> do
+        failAfter 60 $
+          withTempDir "temp-dir-to-check-hydra-logs" $ \dir -> do
+            let hydraSK = dir </> "hydra.sk"
+            hydraSKey :: SigningKey HydraKey <- generate arbitrary
+            void $ writeFileTextEnvelope hydraSK Nothing hydraSKey
+            bracket
+              (createProcess (proc "hydra-node" ["-n", "hydra-node-1", "--hydra-signing-key", hydraSK]){std_out = CreatePipe})
+              cleanupProcess
+              ( \(_, Just nodeOutput, _, _) -> do
+                  out <- hGetLine nodeOutput
+                  let loadedOptions = Aeson.encode $ out ^? key "message" . key "node" . key "runOptions"
+
+                      expectedPort = Just $ Aeson.Number 4001
+                      expectedApiHost =
+                        Just $
+                          Aeson.object ["ipv4" .= Aeson.String "127.0.0.1", "tag" .= Aeson.String "IPv4"]
+                      expectedHost =
+                        Just $
+                          Aeson.object ["ipv4" .= Aeson.String "127.0.0.1", "tag" .= Aeson.String "IPv4"]
+                      expectedHydraScript =
+                        Just $ Aeson.String "0101010101010101010101010101010101010101010101010101010101010101"
+                  -- now we can check some of the default values to see if the node logs them
+                  -- NB: Object/HashMap doesn't know about the order of json fields so that is why we are not able
+                  -- to constuct a object and assert it is the same as the one we get from parsing the log
+                  out ^? key "message" . key "node" . key "tag" `shouldBe` Just (Aeson.String "NodeOptions")
```

Comment:
I think this is enough for this test and rather remove the other default values (it's just another test to update when we change them). Rather add a `roundtripGoldenSpec` on the `RunOptions` somewhere and we're good.